=== modified file 'django-nova/src/django_nova/connection.py'
--- django-nova/src/django_nova/connection.py   2011-03-15 06:02:40 +0000
+++ django-nova/src/django_nova/connection.py   2011-04-22 17:14:22 +0000
@@ -23,13 +23,13 @@
 from django.conf import settings


-def get_nova_admin_connection():
+def get_nova_admin_connection(clc_url=None, region=None):
     """
     Returns a Nova administration connection.
     """
     return adminclient.NovaAdminClient (
-        clc_url=settings.NOVA_DEFAULT_ENDPOINT,
-        region=settings.NOVA_DEFAULT_REGION,
+        clc_url=clc_url or settings.NOVA_DEFAULT_ENDPOINT,
+        region=region or settings.NOVA_DEFAULT_REGION,
         access_key=settings.NOVA_ACCESS_KEY,
         secret_key=settings.NOVA_SECRET_KEY
     )

=== modified file 'django-nova/src/django_nova/manager.py'
--- django-nova/src/django_nova/manager.py      2011-03-19 01:48:42 +0000
+++ django-nova/src/django_nova/manager.py      2011-04-22 17:08:35 +0000
@@ -50,7 +50,8 @@
         Returns a buffer of a zip file containing signed credentials
         for the project's Nova user.
         """
-        nova = get_nova_admin_connection()
+        nova = get_nova_admin_connection(clc_url=self.region['endpoint'],
+                                         region=self.region['name'])
         return nova.get_zip(self.username, self.projectname)

     def get_images(self, image_ids=None):
